### ISSUE TITLE
System Environment Variables override Local Environment Variables for Child processes

### ISSUE THREAD

**Issue by adamsd5** [cbucher/console#345]
### Expected Behavior
I expect to see environment variables for Console.exe to pass into the shell process.  In the steps below, I expect **echo %FOO%** to show **FromLocal**

### Actual Behavior
In this test case, the Windows System Environment variable is overriding the variable in Console.exe's environment.  **echo %FOO%** actually shows **FromSystem**

### Steps to reproduce
In Windows Environment variables, add a System environment variable:
FOO=FromSystem

In a Windows Command prompt, type:
set FOO=FromLocal
echo %FOO%
  - Note that the local setting is in the current environment
Console.exe
  - You can use procexp.exe at this point to see the Console.exe process in fact has FOO=FromLocal.  

In the cmd shell that opens in Console.exe's window, type:
echo %FOO%

### Diagnostic Report
```
[conzoleZ Diagnostic Info.txt](https://github.com/cbucher/console/files/260312/conzoleZ.Diagnostic.Info.txt)

(I tried testing with the beta version, but my work-IT blocks your download site.  I also tried building the latest code, which I can do, but I have trouble running it; something related to the Hook... happy to share more and help debug this if you want some help;  If I figure it out, I'll send a pull request.)
```

### Crash Report
No crash report

**Comment by adamsd5** [comment]
I looked through the code a bit, and I found that **CreateEnvironmentBlock** used in **ConsoleHandler::UpdateCurrentUserEnvironmentBlock** returns the system variable, not the one from the Console.exe process.  This seems to disagree with MSDN documentation, but go figure.  However, take a look at **GetEnvironmentStrings()**.  It seems to return the same kind of environment variable block, but the one from the current running process.  This might be the right way to go.
